Stripe/STPPaymentContext.m

```diff
@@ -167,6 +167,25 @@ - (void)setSelectedPaymentMethod:(id<STPPaymentMethod>)selectedPaymentMethod {
     }
 }
 
+- (void)setPaymentSummaryItems:(NSArray<PKPaymentSummaryItem *> *)paymentSummaryItems {
+    FAUXPAS_IGNORED_IN_METHOD(APIAvailability);
+    if (paymentSummaryItems.count > 0) {
+        PKPaymentSummaryItem *lastItem = paymentSummaryItems.lastObject;
+        NSDecimalNumber *amount = [NSDecimalNumber stp_decimalNumberWithAmount:self.paymentAmount
+                                                                      currency:self.paymentCurrency];
+        if ([amount isEqualToNumber:lastItem.amount]) {
+            _paymentSummaryItems = paymentSummaryItems;
+        }
+        else {
+            _paymentSummaryItems = nil;
+            NSAssert(0, @"Amount for last object in paymentSummaryItems does not equal paymentAmount (%@ vs %@, currency = %@)", lastItem.amount, amount, self.paymentCurrency);
```

Comment:
@bdorfman - I know I'm a little late to the party here. Do you think instead of `NSAssert`ing here you could just squash the current value of `paymentAmount` when `paymentSummaryItems` is set? (That way, developers who want to use this will only have to set one property instead of 2, and not worry about keeping them in sync.)